agent.go

```diff
@@ -1295,9 +1306,25 @@ func cgroupsMount() error {
 		}
 	}
 
-	// Enable memory hierarchical account.
-	// For more information see https://www.kernel.org/doc/Documentation/cgroup-v1/memory.txt
-	return ioutil.WriteFile(cgroupMemoryUseHierarchyPath, []byte{'1'}, cgroupMemoryUseHierarchyMode)
+	if !unifiedCgroupHierarchy {
+		// Enable memory hierarchical account.
+		// For more information see https://www.kernel.org/doc/Documentation/cgroup-v1/memory.txt
+		return ioutil.WriteFile(cgroupMemoryUseHierarchyPath, []byte{'1'}, cgroupMemoryUseHierarchyMode)
+	}
+
+	// Enable all cgroup v2 controllers
+	rawControllers, err := ioutil.ReadFile(cgroupControllersPath)
+	if err != nil {
+		return err
+	}
+
+	var controllers string
+	for _, c := range strings.Fields(string(rawControllers)) {
+		controllers += fmt.Sprintf("+%v ", c)
+	}
+
+	return ioutil.WriteFile(cgroupSubtreeControlPath,
```

Comment:
Nit: worth adding a link to the v2 doc for parity with the cgroup v1 URL doc link above?